recipe_scrapers/reishunger.py

```diff
@@ -27,17 +27,15 @@ def ingredients(self):
         return self.schema.ingredients()
 
     def instructions(self):
-        result = self.soup.find("section", {"class": "recipe-preparation"})
-        if result:
-            result = "\n".join(
-                normalize_string(i.get_text()) for i in result.findAll("p")
-            )
-        return result
+        elements = self.soup.findAll("div", {"class": "flex flex-col"})
+
+        # for recipes that do NOT allow to switch between different cooking methods
+        for e in elements:
+            if "Zubereitung" in e.get_text():
+                return e.get_text().replace("\xa0", " ")
```

Comment:
A few items of feedback here:

- The ideal return format for `instructions` is one entry per newine, with no empty newlines, so let's see if we can achieve that
- `div.flex.flex-col` is quite a large selection - 70+ items.  There's probably a more precise and efficient way we can navigate to the text to retrieve
- Checking for the presence of particular words in a part of the page seems potentially fragile - maybe we can avoid that too
- Nitpick: recommend renaming `e` (often used to refer to exception objects) to `element`

Reply:
1. i can change the format to split the text into sentences
2. it is large but it only contains the instructions, everything smaller might not contain them but i can certainly look again
3. there was nothing else on the page and since its a german page it should always be this. 
4. can do that

It most certainly wont be an ideal parser but the current one simply does not work at all so i guess everything is better ... will look into the changes when I have some time the next few days

Reply:
All sounds good, thanks @vabene1111.  In particular: after attempting to code a few other extraction techniques, I agree on point (3) that using `Zubereitung` is acceptable (it's challenging to identify the correct `h3` to find text within otherwise).

Reply:
ok so i fixed the loop. Regarding the instructions I am not sure how you prefer it. I could add a line break basically on every end of a sentence (basically `replace('.', '.\n')` which would be one entry per new line. On the other hand most recipes on reishunger.de are already formatted in logical paragraphs which could be left as a block. 

what do you prefer?

Reply:
Roughly speaking I think the goals should be:

- One `instruction` (line) in our output for each step described by a source website
  - this website may be a special case, because "preparations" instructions contain nested steps.. let's try to flatten those into a single list, because that's what our output format supports
- Empty lines omitted from the output
- Paragraphs usually retained in a single instruction -- splitting every sentence is (usually) too much.

Because of the nested preparations, this site seemed tricky - I had to try coding it to figure out the limitations, and I've pushed 2c1f4efcf0a83141c35f3a182f14998a3326ea36 with the results of that investigation.

Is there a way to preview how this recipe would currently be rendered in Tandoor?

Reply:
Ah, and to explain one particular line changed from the original:

https://github.com/hhursev/recipe-scrapers/pull/616/commits/2c1f4efcf0a83141c35f3a182f14998a3326ea36#diff-71692e58f6958c38c11a66499d988e549e7b00e794c9aa8e430f8714476bf2d8L30

My reason for changing this `find` statement is that originally it was potentially collecting a _lot_ of `div` elements on the page (70 or so, I think).  Iterating over those can be a bit slow.  Unfortunately we don't have a way to benchmark the performance of scrapers yet, but that would be nice to have in future.